testcode/lib/joy-test.c

```diff
@@ -56,23 +74,31 @@ int main (void)
             gotoxy (0, i+1);
             j = joy_read (i);
 #if defined(__ATARI5200__) || defined(__CREATIVISION__)
-            cprintf ("%1d:%-3s%-3s%-3s%-3s%-3s%-3s",
+            cprintf ("%1d:%-3s%-3s%-3s%-3s%-3s %02x",
                      i,
                      JOY_UP(j)?    " U " : " - ",
                      JOY_DOWN(j)?  " D " : " - ",
                      JOY_LEFT(j)?  " L " : " - ",
                      JOY_RIGHT(j)? " R " : " - ",
-                     JOY_BTN_1(j)? " 1 " : " - ");
+                     JOY_BTN_1(j)? " 1 " : " - ", j);
 #else
-            cprintf ("%2d: %-6s%-6s%-6s%-6s%-6s%-6s",
+            cprintf ("%2d: %-6s%-6s%-6s%-6s%-6s %02x",
                      i,
                      JOY_UP(j)?    "  up  " : " ---- ",
                      JOY_DOWN(j)?  " down " : " ---- ",
                      JOY_LEFT(j)?  " left " : " ---- ",
                      JOY_RIGHT(j)? "right " : " ---- ",
-                     JOY_BTN_1(j)? "button" : " ---- ");
+                     JOY_BTN_1(j)? "button" : " ---- ", j);
 #endif
         }
+
+        /* show pressed key, so we can verify keyboard is working */
+        kb = kbhit();
```

Comment:
Please insert a space in front of the parameter parenthesis here and on line 97, 99, 100 and 101.